src/main/java/com/commercetools/sync/commons/utils/SyncSolutionInfo.java

```diff
@@ -3,7 +3,7 @@
 import io.sphere.sdk.client.SolutionInfo;
 
 public final class SyncSolutionInfo extends SolutionInfo {
-    private static final String LIB_NAME = "commercetools-sync-java";
+    public static final String LIB_NAME = "commercetools-sync-java";
```

Comment:
As we are targeting the add version header within the CustomHeaderSphereClientDecorator, do we need to remove the solution info metafile from the resources? WDYT? 

https://github.com/commercetools/commercetools-sync-java/blob/master/src/main/resources/META-INF/services/io.sphere.sdk.client.SolutionInfo

Reply:
I have seen Hesham's PR and the metafile is created at that moment
https://github.com/commercetools/commercetools-sync-java/issues/142

On the other hand, I found now the library version injection is done by this, which is added by Hesham later.
https://github.com/commercetools/commercetools-sync-java/blob/251-custom-decorator-fix-test/gradle-scripts/set-library-version.gradle

@butenkor Can you confirm that the solution info metafile is deprecated already?

Reply:
First of all, our SyncSolutionInfo might stay as you will need the version in Gradle, also the Gradle script is make sense to automate getting the latest version of the library.

The work here is started to create an alternative of SolutionInfo for adding the version headers using decorators. Now it looks that we have it.

Here, I would remove the solution info metafile, and also remove the extending SolutionInfo of the SyncSolutionInfo, and run an integration test and ensure the new headers in the API.

The patch: 
````
Index: src/main/java/com/commercetools/sync/commons/utils/SyncSolutionInfo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/commercetools/sync/commons/utils/SyncSolutionInfo.java	(revision 471f863328aafbcb9326be171ae97ae64d60e30b)
+++ src/main/java/com/commercetools/sync/commons/utils/SyncSolutionInfo.java	(date 1606755014333)
@@ -1,25 +1,9 @@
 package com.commercetools.sync.commons.utils;
 
-import io.sphere.sdk.client.SolutionInfo;
-
-public final class SyncSolutionInfo extends SolutionInfo {
+public final class SyncSolutionInfo {
     public static final String LIB_NAME = "commercetools-sync-java";
     /**
      * This value is injected by the script at gradle-scripts/set-library-version.gradle.
      */
     public static final String LIB_VERSION = "#{LIB_VERSION}";
-
-    /**
-     * Extends {@link SolutionInfo} class of the JVM SDK to append to the User-Agent header with information of the
-     * commercetools-sync-java library
-     *
-     * <p>A User-Agent header with a solution information looks like this:
-     * {@code commercetools-jvm-sdk/1.46.1 (AHC/2.0) Java/1.8.0_92-b14 (Mac OS X; x86_64)
-     * {@value LIB_NAME}/{@value LIB_VERSION}}</p>
-     *
-     */
-    public SyncSolutionInfo() {
-        setName(LIB_NAME);
-        setVersion(LIB_VERSION);
-    }
-}
\ No newline at end of file
+}
diff --git src/main/resources/META-INF/services/io.sphere.sdk.client.SolutionInfo src/main/resources/META-INF/services/io.sphere.sdk.client.SolutionInfo
deleted file mode 100644
index 7ade83f6f85b957eff000084d8b2940c727c4820..0000000000000000000000000000000000000000
GIT binary patch
literal 0
Hc$@<O00001
`````

Reply:
I just run IT locally and check the logs in Kibana.
Inheritance from SolutionInfo is a must, because it provides setter method to SyncSolutionInfo to set library name and version.

I tried to remove the metafile and run the test. The library name and version immediately disappear in User-Agent in request header log. For this reason i would keep the metafile.

Reply:
>Inheritance from SolutionInfo is a must, because it provides setter method to SyncSolutionInfo to set library name and version.

So the solution that you have found as a custom decorator not working ? 🤔 Please check the old discussions of this work, I would expect without a metafile and solution info we should be adding headers 🤔